src/Build/Graph/ProjectGraph.cs

```diff
@@ -317,14 +307,105 @@ public ProjectGraph(ProjectGraphEntryPoint entryPoint, ProjectCollection project
             IEnumerable<ProjectGraphEntryPoint> entryPoints,
             ProjectCollection projectCollection,
             ProjectInstanceFactoryFunc projectInstanceFactory)
+            : this(
+                entryPoints,
+                projectCollection,
+                projectInstanceFactory,
+                Environment.ProcessorCount - 1,
```

Comment:
>Environment.ProcessorCount - 1, [](start = 16, length = 31)

Why -1 here but not in the overload below? Also, why -1 here?

Reply:
Should be ProcessorCount, forgot to update this overload

Reply:
Well actually, should it be -1 to account for the current thread that runs the constructor? It will participate in the graph constructor too, so -1 ensures thread count matches processor count.

Reply:
Oh, you also decrement 1 also when instantiating the ParallelWorkSet, so yeah, no -1 here. Maybe replace -1 with a well named constant :)

Reply:
That's over in the GraphBuilder constructor:
https://github.com/microsoft/msbuild/blob/78d3a5b134554f157b690b38dec53492a50a587c/src/Build/Graph/GraphBuilder.cs#L50-L54